x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/AsyncSearchUser.java

```diff
@@ -16,7 +16,8 @@
     public static final AsyncSearchUser INSTANCE = new AsyncSearchUser();
     public static final String ROLE_NAME = UsernamesField.ASYNC_SEARCH_ROLE;
     public static final Role ROLE = Role.builder(new RoleDescriptor(ROLE_NAME,
-            null,
+            // handles template registry for the .async-search index
+            new String[] { "manage_index_templates" },
```

Comment:
Hmm, does this meant that any async search user can now mess with templates. Does this really necessary?